### ISSUE TITLE
Weird behavior for objects using TS4.9 `satisfies` keyword

### ISSUE THREAD

**Issue by blixt** [swc-project/jest#130]
I ran into a very strange error that took me a while to narrow down, but I finally figured out it's `@swc/jest` (or `@swc/core`) failing to handle the new `satisfies` keyword in TypeScript 4.9.

I made a repro here: https://github.com/blixt/test-swc-jest-bug (run `yarn && yarn jest` to see the error)

<img width="696" alt="Screenshot 2022-11-25 at 01 46 31" src="https://user-images.githubusercontent.com/158591/203879515-8b5b6a50-dae3-4061-a0b5-30269d6be064.png">

In short, the bug is that values imported from other modules are not found by `swc` when they are referenced inside of an object that has the `satisfies` operator. It's as if the object becomes a scope that is not able to access the imported values. Values in the same module can be referenced, however.

**Comment by blixt** [comment]
I verified it now works with the latest versions of all packages.
